docs/src/documentation/03-components/04-overlay/tooltip/03-accessibility.mdx

```diff
@@ -0,0 +1,84 @@
+---
+title: Accessibility
+redirect_from:
+  - /components/tooltip/accessibility/
+---
+
+# Accessibility
```

Comment:
I will post my comment here because it is connected to multiple places. I am super confused what is the purpose of the `aria-label` and `aria-labelledby` props in this component so I have a lot of questions 🙈 

1. In the current implementation, `TooltipWrapper` has the role `button`, is that correct? Do you know why it has this role? I was trying to use screen reader and the "button" announce was always very confusing.

2. If there is a text provided as a children, there are no a11y violations - is it enough like this or should the `aria-label` (or `aria-labelledby`) be provided anyway, maybe with prefix "tooltip", that there is a tooltip? All of the violations with tooltip I saw were caused by the triggering (reference) element being icon - is this one of the examples where should the `aria-label` (or `aria-labelledby`) be provided or it should really be provided always, e.g. to avoid (double) announce of the content of reference element (I am kinda repeating myself here but you get what I am trying to ask 😅)? Should this be mentioned? The double announce happened to me on booking page with the self-transfer note, will we fix this?

3. Have you take a look at the [aria-details](https://www.w3.org/TR/wai-aria/#aria-details) property? Is it something we should also add here? If I understand correctly, it might be useful when there is more complex content inside the tooltip (e.g. lists) so the content of the element referenced by `aria-details` is not flattened to a string when presented to assistive technology users. I am asking as that seems to be a common usage of tooltips across our pages.

![img](https://github.com/user-attachments/assets/c5cd0e86-96f6-4bce-9b8d-eb92b020fafb)

5. Two of three examples are the same, maybe one can be removed? I am not sure what should be the correct behavior with the children and what we need to fix, but if the `aria-label` (or `aria-labelledby`) are not provided and children is text, component is already accessible (or at least it doesn't violate anything) and these props don't need to be added. Should we mention that or is that just adding the expected things to the documentation (and what Daniel said is a bad precedent)? Maybe we could mention what should be passed to `aria-label` ideally or how should the element(s) being referenced by the `aria-labelledby` look like - should it sum up the content of the tooltip? Or just make introduction to tooltip content? Maybe the example can be real (as in the Seat component documentation where it mentions seat for a specific passenger) instead of just lorem ipsum text. Current examples also suggests that the `aria-labelledby` needs to be connected with the reference element but that't not true, no? Maybe we can use it with different element than children in the example?

Reply:
1. I've found that the [triggering component should be interactive](https://sarahmhigley.com/writing/tooltips-in-wcag-21/) - it was written on more websites, but I can't find it right now. I've found some sources talking about the `<a>` element, but AFAIK the `role=button` should be a general role for this. On the other side, we don't know what devs (will) put inside as a children component, so it might escalate other violation issues (?) -
Example: in case there will be a `<button>` element as children, then we can get the `Interactive controls must not be nested`.
 
2. I think it was discussed above in the comment. ⏫ 

3. No, I haven't checked this aria-details property, tbh. It's hard to say if it makes sense to add this prop, on the other side, according to the articles about tooltips, the tooltip content should be simple, without any images/links/whatever, on the other side, we have various content in our tooltips across the Kiwi pages. 🤔 
I'm confused and lost about which approach should be better, tbh.

4. Yes, the two examples are the same, as I wanted to describe the further behavior. But I can take a look at it, and polish it, ofc.

Reply:
Regarding point 1., I believe the `role="button"` is present because on mobile it is actually a clickable element, to trigger the MobileDialog. If that is the most correct approach…? I honestly don't know 😓 probably the role should be added only when it is indeed expecting a click. However, that would not prevent the raised question: what happens if one uses button as a tooltip trigger (should one even do it, ever?)